api/enki.py

```diff
@@ -0,0 +1,542 @@
+    Configuration,
+    temp_config,
+)
+
+from circulation import (
+    LoanInfo,
+    FulfillmentInfo,
+    HoldInfo,
+    BaseCirculationAPI
+)
+
+from circulation_exceptions import *
+
+from core.util import LanguageCodes
+from core.util.http import (
+    HTTP,
+    RemoteIntegrationException,
+)
+
+from core.coverage import (
+    BibliographicCoverageProvider,
+    CoverageFailure,
+)
+
+from core.model import (
+    get_one_or_create,
+    Collection,
+    Contributor,
+    DataSource,
+    DeliveryMechanism,
+    LicensePool,
+    Edition,
+    Identifier,
+    Library,
+    Representation,
+    Subject,
+    ExternalIntegration,
+)
+
+from core.metadata_layer import (
+    SubjectData,
+    ContributorData,
+    FormatData,
+    IdentifierData,
+    CirculationData,
+    Metadata,
+    ReplacementPolicy,
+)
+
+from core.monitor import (
+    Monitor,
+    IdentifierSweepMonitor,
+    CollectionMonitor,
+)
+
+from core.opds_import import SimplifiedOPDSLookup
+from core.analytics import Analytics
+
+#TODO: Remove unnecessary imports (once the classes are more or less complete)
+
+class EnkiAPI(BaseCirculationAPI):
+
+    DESCRIPTION = _("Integrate an Enki collection.")
+    SETTINGS = [
+        { "key": Collection.EXTERNAL_ACCOUNT_ID_KEY, "label": _("Library ID") },
+        { "key": ExternalIntegration.URL, "label": _("URL") },
+    ] + BaseCirculationAPI.SETTINGS
+
+    PRODUCTION_BASE_URL = "http://enkilibrary.org/API/"
+    availability_endpoint = "ListAPI"
+    item_endpoint = "ItemAPI"
+    #TODO: get this from configurationsettings. Also make sure the value actually gets to the DB.
+    lib = 1
+
+    NAME = u"Enki"
+    ENKI = NAME
+    ENKI_EXTERNAL = NAME
+    ENKI_ID = u"Enki ID"
+
+    SET_DELIVERY_MECHANISM_AT = BaseCirculationAPI.BORROW_STEP
+    SERVICE_NAME = "Enki"
+    log = logging.getLogger("Enki API")
+    log.setLevel(logging.DEBUG)
+
+    def __init__(self, _db, collection):
+        self._db = _db
+
+        if collection.protocol != self.ENKI:
+            raise ValueError(
+                "Collection protocol is %s, but passed into EnkiAPI!" %
+                collection.protocol
+            )
+        self.collection = collection
+        self.library_id = collection.external_account_id.encode("utf8")
+        self.base_url = collection.external_integration.url or self.PRODUCTION_BASE_URL
+
+        if not self.library_id or not self.base_url:
+            raise CannotLoadConfiguration(
+                "Enki configuration is incomplete."
+            )
+        self.enki_bibliographic_coverage_provider = (
+            EnkiBibliographicCoverageProvider(
+                _db, collection, api_class=self
+            )
+        )
+        self.token = "mock_token"
+
+    @property
+    def authorization_headers(self):
+        # I don't think Enki needs authorization headers...
+        #authorization = u":".join([self.username, self.password, self.library_id])
+        authorization = u":".self.library_id
```

Comment:
This will cause a crash, but it's called by `refresh_bearer_token`, which makes me think the whole token code is broken.